### DISCUSSION TITLE
Validate BootStrap Markup and CSS on Website

### DISCUSSION THREAD

**Discussion by jacobmedley** [twbs/bootstrap#38875] [Q&A]
We have several 3rd party developers that have been breaking away from BootStraps patterns for markup and css. 

Is there a tool that can scan a website and report on when “bad” bootstrap code is present?

An example would be the card component. css was changed from card-body to card-block and custom css was written to get the same result.

**Comment by patkon** [comment]
I'm unfortunately not able to find an answer for this use case. But I would also like it if there is a tool which could output / compare all classes of a project.

Without knowing your code I would guess the following: The developer used the alpha BS4 (there was `.card-block` included). After an update to a newer version he didn't want to change (or could not) the HTML markup and copied the class `.card-body. This is not nice and saves (developer's) time at the expense of performance.

**Comment by ishan-18** [comment]
You can use the `Bootlint` tool to scan your website's HTML files and detect issues with Bootstrap usage. Install Bootlint using npm and run it with the command `bootlint your-html-file.html`, replacing your-html-file.html with the path to the HTML file you want to check. This will help you identify common mistakes and deviations from Bootstrap's guidelines in the HTML code.

Please note that Bootlint focuses on HTML usage, and for CSS-related issues, you may need to use a separate CSS linting tool like `stylelint` to enforce Bootstrap guidelines and detect deviations in CSS code.

For bootstrap, you can use `stylelint-config-twbs-bootstrap` package.
